mypy/main.py

```diff
@@ -416,6 +424,100 @@ def get_init_file(dir: str) -> Optional[str]:
     return None
 
 
+# For most options, the type of the default value set in options.py is
+# sufficient, and we don't have to do anything here.  This table
+# exists to specify types for values initialized to None or container
+# types.
+config_types = {
+    # TODO: Check validity of python version
+    'python_version': lambda s: tuple(map(int, s.split('.'))),
+    'strict_optional_whitelist': lambda s: s.split(),
+    'custom_typing_module': str,
+}
+
+
+def parse_config_file(options: Options, filename: str) -> None:
+    """Parse a config file into an Options object.
+
+    Errors are written to stderr but are not fatal.
+    """
+    parser = configparser.RawConfigParser()
+    try:
+        parser.read(filename)
+    except configparser.Error as err:
+        print("%s: %s" % (filename, err), file=sys.stderr)
+        return
+    if 'mypy' not in parser:
+        print("%s: No [mypy] section in config file" % filename, file=sys.stderr)
+        return
+
+    section = parser['mypy']
+    prefix = '%s: [%s]' % (filename, 'mypy')
+    updates, report_dirs = parse_section(prefix, options, section)
+    for k, v in updates.items():
+        setattr(options, k, v)
+    options.report_dirs.update(report_dirs)
+
+    for name, section in parser.items():
+        if name.startswith('mypy-'):
```

Comment:
Now sections are a little limited. I'd like to be able to specify multiple globs per section to avoid repetition. Here is a potential syntax:

```
[mypy dir/*.py, dir2/*.py]
...
```

(Not sure if configparser supports this syntax.)

Also, it's unclear if a directory is a valid glob. I think that it should be -- I'd expect `[mypy dir]` to apply to all files under `dir/` if it's a directory.